pkg/controller.v2/controller_status.go

```diff
@@ -84,6 +107,103 @@ func (tc *TFJobController) updateStatus(tfjob *tfv1alpha2.TFJob, rtype tfv1alpha
 	return nil
 }
 
+func (tc *TFJobController) updateStatusNew(tfjob *tfv1alpha2.TFJob, rstatus map[string]v1.PodPhase, wreplicas int, preplicas int) error {
```

Comment:
Please add a comment explaining what this function is doing. In particular, if its updating status please describe the logic for determining the status.

Reply:
@jlewi I will add it as soon as possible